Sawyern

sawyern opened the pull request:
Hi Jenica,

Looks great visually, but I tried my best to clean stuff up and fix your bugs.
Some general notes here, then check the comments I added for more details.

Feel free to merge this with yours, or take what you need cherry-pick style. Up to you.

1. I added a Globals file, separate from your main to store any variables you want exposed globally. I also added it to Project -> Project Settings -> AutoLoad. This way you don't have to rename any files it is good as is.
2. I removed all your extends from main. You went a little crazy here with the extensions, but this isn't the intended effect you wanted. It was erratically duplicating a lot of code and causing a lot of issues (ie. double printing of loaded scenes, this is how I was able to tell and track it down. Printing is king good job). I changed them all back to Node2D.
3. Some of your bugs were caused by the fact that you had so many levels copy pasted, but only fixed a couple of them. Just be careful what you copy paste and make sure you update your fixes everywhere. In general, there is probably a cleaner solution where you only have one file (one change, one place), but this is fine for a beginner project, just try to make sure you make one perfect before you copy paste many times.
4. Check the links on your help buttons. I think its not hooked up to the function properly, which is why you are running into issues on certain levels.
5. I added the Github plugin free of charge!
6. Feel free to ask questions. I'm here to help :)

sawyern committed:
add git plugin

sawyern committed:
added various notes and bug fixes, see PR notes

sawyern committed:
See notes

sawyern commented on Scenes/_0main.tscn:
```diff
@@ -0,0 +1,13 @@
+[gd_scene load_steps=2 format=2]
```
I created this while testing and forgot to delete. Feel free to delete this file.

sawyern commented on Scenes/disabled.gd:
```diff
@@ -1,11 +1,10 @@
 func _ready():
 	pass # Replace with function body.
 
 func _activate_level_1():
```
You can simplify this as well.

`$self.visible = Globals.level1_complete`

jenicat reviewed (approved):
Thanks for spending so much time working on this messy code. I could tell that copying and pasting was a problem, and thanks for helping me figure out how the Globals work.

sawyern commented:
> Thanks for spending so much time working on this messy code. I could tell that copying and pasting was a problem, and thanks for helping me figure out how the Globals work.

No problem. If you click the "merge pull request" button, it will merge with your existing code automatically. It's up to you if you'd like to do that or not.

Sawyer

jenicat committed:
Finished, hopefully.

I activated the levels, and installed a tutorial.

I didn't get to adding levels 7 - 10, but this is good for now.

jenicat committed:
Retry

I don't think the last upload went through
